themes/theme-ios11/pages/Search/components/Content/index.jsx

```diff
@@ -38,16 +44,19 @@ class SearchContent extends Component {
    */
   render() {
     const {
-      searchPhrase, showFilterBar, showNoResults,
+      searchPhrase, showFilterBar, showNoResults, pattern,
     } = this.props;
 
     return (
       <RouteContext.Consumer>
         {({ state, query, id: routeId }) => (
           <Fragment>
-            <BackBar title={searchPhrase} shadow={!showFilterBar} />
+            <BackBar
+              title={pattern === CATEGORY_ALL_PATTERN ? state.categoryName : searchPhrase}
+              shadow={!showFilterBar}
+            />
 
-            {showFilterBar && <Bar key="below" /> }
+            {showFilterBar && <Bar pattern={pattern} key="below" /> }
```

Comment:
as far as i can see, the `Bar` component already includes logic to access the current route pattern.